docs/data-sources/service_principal.md

```diff
@@ -37,7 +37,8 @@ Data source allows you to pick service principals by the following attributes
 
 Data source exposes the following attributes:
 
-- `sp_id` - The id of the service principal.
+- `id` - The id of the service principal.
+- `sp_id` - The id of the service principal (deprecated, kept for compatibility with earlier versions).
```

Comment:
probably just remove it from docs. Also please mention this in the PR description.